### PR TITLE
Remove dap_prototype

### PR DESCRIPTION
**Pull request by mendess**

### CODE REVIEW

**Comment by mendess** [comment]
I deleted the docker build step because:
- we no longer have a worker implementation to deploy to docker
- it's been broken in main for months

**Comment by cjpatton** [comment]
🖖

**Review by cjpatton — APPROVED** [review]
Looks good to me! I left a few comments for things to double check. In one spot in particular, I think we want to gracefully handle the error instead of panicking.

Just a heads up that I took the liberty of rebasing so that I can start stacking on this PR. The previous head was 3a90b709a088e633f3612a8aae40e950716592b0.

Also, `cargo test` currently fails, though `cargo test --all-targets` succeeds. I think this suggests the default feature set needs to be cleaned up:
```
   Doc-tests daphne_worker

running 1 test
test daphne_worker/src/storage_proxy.rs - storage_proxy (line 41) ... FAILED

failures:

---- daphne_worker/src/storage_proxy.rs - storage_proxy (line 41) stdout ----
error[E0432]: unresolved import `reqwest_wasm`
 --> daphne_worker/src/storage_proxy.rs:43:5
  |
4 | use reqwest_wasm as reqwest;
  |     ^^^^^^^^^^^^^^^^^^^^^^^ no external crate `reqwest_wasm`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
Couldn't compile the test.

failures:
    daphne_worker/src/storage_proxy.rs - storage_proxy (line 41)

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.10s

error: doctest failed, to rerun pass `-p daphne_worker --doc`
```

**Comment by cjpatton** [comment]
@mendess I've tacked on a couple of commits to make CI happy. PTAL, in particular https://github.com/cloudflare/daphne/pull/497/commits/9d52eda4eb659118e9d6cda0eb1fed03982e7eec.

File: daphne_worker/src/durable/mod.rs [hunk 1]
```diff
@@ -62,39 +62,6 @@ impl<'srv> DurableConnector<'srv> {
         DurableConnector { env, retry: false }
     }
```
[the remark this answers is not in the document]
Comment (mendess) [reply daphne_worker/src/durable/mod.rs:64]: They were already implemented in `daphne_server/src/storage_proxy_connection/mod.rs`

File: daphne_worker/src/durable/mod.rs [hunk 2]
```diff
@@ -562,16 +525,6 @@ impl DaphneWorkerDurableConfig {
             }
         };
 
-        let report_storage_max_future_time_skew = env
```
[the remark this answers is not in the document]
Comment (mendess) [reply daphne_worker/src/durable/mod.rs:565]: This was a mistake. It should have never been in this struct as it is actually used by the `aggregator` logic to reject reports in `DapReportInitializer`

File: daphne_worker/src/lib.rs [hunk 3]
```diff
@@ -194,12 +191,16 @@ pub(crate) fn now() -> u64 {
...
-        Ok("storage-proxy") => DapWorkerMode::StorageProxy,
-        _ => DapWorkerMode::DapPrototype,
+    match env
+        .var("DAP_WORKER_MODE")
+        .expect("DAP_WORKER_MODE is required")
+        .to_string()
+        .as_str()
+    {
+        "storage-proxy" => DapWorkerMode::StorageProxy,
+        invalid => panic!("invalid DAP_WORKER_MODE: {invalid}"),
```
[the remark this answers is not in the document]
Comment (mendess) [reply daphne_worker/src/lib.rs:204]: A panic translates into a 500 for the client and a log message for us. I think this is acceptable as a misconfigured service should fail as soon as possible. There is no possible graceful handling besides logging the error and returning 500, which this already does.

**Pull request merged**